src/extensions/default/CodeFolding/main.js

```diff
@@ -348,5 +422,28 @@ define(function (require, exports, module) {
         }
     }
 
-    AppInit.appReady(init);
+    /**
+      * Register change listener for the preferences file.
+      */
+    function watchPrefsForChanges() {
+        prefs.prefBase.on("change", function (e, data) {
+            if (data.ids.indexOf("enabled") > -1) {
+                if (prefs.getSetting("enabled")) {
+                    init();
+                } else {
+                    var editor = EditorManager.getCurrentFullEditor();
+                    if (editor && editor._codeMirror && CodeMirror.commands.unfoldAll) {
+                        CodeMirror.commands.unfoldAll(editor._codeMirror);
+                        foldGutter.remove(editor._codeMirror);
+                    }
```

Comment:
It will definitely look bad for split view, where two editors are visible at once even though only one is active.

The cleanest way to do it is walking the `_instances` array in Editor.js, which is currently inaccessible -- that's how core Editor preferences are updated.  It seems fair to me to add a public API like `forEveryEditor()` or something -- pass a callback, and it gets invoked like a forEach.